src/main/java/cooking_converter/EValidationErrors.java

```diff
@@ -0,0 +1,23 @@
+package cooking_converter;
+
+public enum EValidationErrors {
+    ERR_AMOUNT_NEGATIVE("ERROR. Amount can't be negative. Please enter valid amount."),
+    ERR_INVALID_NUMBER("ERROR. Invalid number. Please enter valid number."),
+    ERR_INVALID_UNIT("ERROR. Unit isn't valid. Please enter valid unit. %s"),
+    ERR_NOT_WEIGHT_UNIT("ERROR. Unit isn't weight unit. Please enter weight unit. %s"),
+    ERR_NOT_VOLUME_UNIT("ERROR. Unit isn't volume unit. Please enter volume unit. %s");
```

Comment:
Interesting way of doing things. Though I would keep it simple and have unique exceptions for each error.

Reply:
This doesn't really leave you with an option of handling a specific error.